src/components/user/user.service.ts

```diff
@@ -588,6 +595,66 @@ export class UserService {
     };
   }
 
+  async listPartners(
+    userId: string,
+    input: PartnerListInput,
+    session: ISession
+  ): Promise<SecuredPartnerList> {
+    const query = this.db
+
+      .query()
+      .match([requestingUser(session), ...permissionsOfNode('Partner')])
+      .optionalMatch([
+        node('requestingUser'),
+        relation('in', '', 'member'),
+        node('', 'SecurityGroup'),
+        relation('out', '', 'permission'),
+        node('canRead', 'Permission', {
+          property: 'partner',
+          read: true,
+        }),
+      ])
+      .return({
+        canRead: [{ read: 'canRead', edit: 'canEdit' }],
+      });
+    let user;
+    try {
+      user = await query.first();
+    } catch (exception) {
+      this.logger.error(`Could not find partners`, {
+        exception,
+        userId: session.userId,
+      });
+      throw new ServerException('Could not find partner', exception);
+    }
+    if (!user) {
+      throw new NotFoundException('Could not find user', 'userId');
+    }
+
+    //TODO: add back in after the partner permission nodes are available in the DB
```

Comment:
Needs the partner permission nodes added: https://github.com/SeedCompany/cord-api-v3/issues/1465

Reply:
added `partners` to procedure code.